src/gui/sidebar/previews/layer/SidebarPreviewLayers.cpp

```diff
@@ -179,6 +245,37 @@ void SidebarPreviewLayers::layerSelected(size_t layerIndex) {
     // Layers are in reverse order (top index: 0, but bottom preview is 0)
     lc->switchToLay(this->previews.size() - layerIndex - 1);
     updateSelectedLayer();
+
+    int actions = 0;
+
+    const auto bgIndex = this->lc->getLayerCount();
+    const auto bottomLayerIndex = bgIndex - 1;
+
+    // if we are above the bottom layer, we can merge down
+    if (layerIndex < bottomLayerIndex) {
+        actions |= SIDEBAR_ACTION_MERGE_DOWN;
+    }
+
+    // if we are above the bottom layer, we can move down
+    if (layerIndex < bottomLayerIndex) {
+        actions |= SIDEBAR_ACTION_MOVE_DOWN;
+    }
+
+    if (layerIndex > 0 && layerIndex != bgIndex) {
+        actions |= SIDEBAR_ACTION_MOVE_UP;
+    }
+
+    // if we are not on the background layer, we can duplicate (aka copy) and delete
+    if (layerIndex < bgIndex) {
+        actions |= SIDEBAR_ACTION_COPY;
+        actions |= SIDEBAR_ACTION_DELETE;
+    }
+
+    gtk_widget_set_sensitive(this->contextMenuMoveUp, actions & SIDEBAR_ACTION_MOVE_UP);
+    gtk_widget_set_sensitive(this->contextMenuMoveDown, actions & SIDEBAR_ACTION_MOVE_DOWN);
+    gtk_widget_set_sensitive(this->contextMenuMergeDown, actions & SIDEBAR_ACTION_MERGE_DOWN);
+    gtk_widget_set_sensitive(this->contextMenuDuplicate, actions & SIDEBAR_ACTION_COPY);
+    gtk_widget_set_sensitive(this->contextMenuDelete, actions & SIDEBAR_ACTION_DELETE);
```

Comment:
There might be a bug here! In my tests, none of the menus are desensitized when I'm using the **Layer Stack Preview**. It works fine with the other preview pane.

![None of the background page's menus are desensitized](https://user-images.githubusercontent.com/46334387/140822293-d9db95d9-a027-40d0-9aec-4b44a4d900c7.png)

![Move first page up menu isn't desensitized](https://user-images.githubusercontent.com/46334387/140822362-6c8ce48c-55de-4fb1-817f-296deb7d7fc1.png)

Reply:
You are correct; they're apparently only sensitized based on the selection in the layer preview. There are also some failed assertions `gtk_widget_set_sensitive: assertion 'GTK_IS_WIDGET (widget)' failed`.

Reply:
This issue is fixed by implementing the context menu as its own class such that the layer sidebars each have a pointer to the same context menu instance (`SidebarLayersContextMenu`).